Mathlib/CategoryTheory/Enriched/FunctorCategory.lean

```diff
@@ -335,4 +346,115 @@ noncomputable def isLimitConeFunctorEnrichedHom :
+    erw [end_.lift_π]
+    rw [enrichedId_π]
+    dsimp
+
+/-- The composition for the `J ⥤ V`-enrichment of the category `J ⥤ C`. -/
+@[simps]
+noncomputable def functorEnrichedComp [HasFunctorEnrichedHom V F₁ F₂]
+    [HasFunctorEnrichedHom V F₂ F₃] [HasFunctorEnrichedHom V F₁ F₃] :
+    functorEnrichedHom V F₁ F₂ ⊗ functorEnrichedHom V F₂ F₃ ⟶ functorEnrichedHom V F₁ F₃ where
+  app j := enrichedComp V _ _ _
+  naturality j j' f := by
+    dsimp
+    ext k
+    dsimp
+    rw [assoc, assoc, enrichedComp_π]
+    dsimp
+    rw [← tensor_comp_assoc]
+    erw [end_.lift_π, end_.lift_π, end_.lift_π]
+    rw [enrichedComp_π]
+    dsimp
+
+@[reassoc (attr := simp)]
+lemma functorEnriched_id_comp [HasFunctorEnrichedHom V F₁ F₂] [HasFunctorEnrichedHom V F₁ F₁] :
+    (λ_ (functorEnrichedHom V F₁ F₂)).inv ≫
+      functorEnrichedId V F₁ ▷ functorEnrichedHom V F₁ F₂ ≫
+        functorEnrichedComp V F₁ F₁ F₂ = 𝟙 (functorEnrichedHom V F₁ F₂) := by aesop_cat
+
+@[reassoc (attr := simp)]
+lemma functorEnriched_comp_id [HasFunctorEnrichedHom V F₁ F₂] [HasFunctorEnrichedHom V F₂ F₂] :
+    (ρ_ (functorEnrichedHom V F₁ F₂)).inv ≫
+      functorEnrichedHom V F₁ F₂ ◁ functorEnrichedId V F₂ ≫
+        functorEnrichedComp V F₁ F₂ F₂ = 𝟙 (functorEnrichedHom V F₁ F₂) := by aesop_cat
+
+@[reassoc]
+lemma functorEnriched_assoc [HasFunctorEnrichedHom V F₁ F₂] [HasFunctorEnrichedHom V F₂ F₃]
+    [HasFunctorEnrichedHom V F₃ F₄] [HasFunctorEnrichedHom V F₁ F₃]
+    [HasFunctorEnrichedHom V F₂ F₄] [HasFunctorEnrichedHom V F₁ F₄] :
+    (α_ _ _ _).inv ≫ functorEnrichedComp V F₁ F₂ F₃ ▷ functorEnrichedHom V F₃ F₄ ≫
+      functorEnrichedComp V F₁ F₃ F₄ =
+        functorEnrichedHom V F₁ F₂ ◁ functorEnrichedComp V F₂ F₃ F₄ ≫
+          functorEnrichedComp V F₁ F₂ F₄ := by
+  ext j
+  dsimp
+  rw [enriched_assoc]
+
+variable (J C) in
+/-- If `C` is a `V`-enriched ordinary category, and `C` has suitable limits,
+then `J ⥤ C` is also a `J ⥤ V`-enriched ordinary category. -/
+noncomputable def functorEnrichedCategory
+    [∀ (F₁ F₂ : J ⥤ C), HasFunctorEnrichedHom V F₁ F₂] :
+    EnrichedCategory (J ⥤ V) (J ⥤ C) where
+  Hom F₁ F₂ := functorEnrichedHom V F₁ F₂
+  id F := functorEnrichedId V F
+  comp F₁ F₂ F₃ := functorEnrichedComp V F₁ F₂ F₃
+  assoc F₁ F₂ F₃ F₄ := functorEnriched_assoc V F₁ F₂ F₃ F₄
+
+variable {F₁ F₂} in
+/-- Given functors `F₁` and `F₂` in `J ⥤ C`, where `C` is a `V`-enriched ordinary category,
+this is the bijection `(F₁ ⟶ F₂) ≃ (𝟙_ (J ⥤ V) ⟶ functorEnrichedHom V F₁ F₂)`. -/
+noncomputable def functorHomEquiv [HasFunctorEnrichedHom V F₁ F₂] [HasEnrichedHom V F₁ F₂] :
+    (F₁ ⟶ F₂) ≃ (𝟙_ (J ⥤ V) ⟶ functorEnrichedHom V F₁ F₂) :=
+  (homEquiv V).trans (isLimitConeFunctorEnrichedHom V F₁ F₂).homEquiv
+
+lemma functorHomEquiv_id [HasFunctorEnrichedHom V F₁ F₁] [HasEnrichedHom V F₁ F₁] :
+    (functorHomEquiv V) (𝟙 F₁) = functorEnrichedId V F₁ := by
+  ext j
+  dsimp [functorHomEquiv]
+  erw [IsLimit.homEquiv_apply]
+  aesop_cat
```

Comment:
```suggestion
@[simps!]
noncomputable def functorHomEquiv [HasFunctorEnrichedHom V F₁ F₂] [HasEnrichedHom V F₁ F₂] :
    (F₁ ⟶ F₂) ≃ (𝟙_ (J ⥤ V) ⟶ functorEnrichedHom V F₁ F₂) :=
  (homEquiv V).trans (isLimitConeFunctorEnrichedHom V F₁ F₂).homEquiv

lemma functorHomEquiv_id [HasFunctorEnrichedHom V F₁ F₁] [HasEnrichedHom V F₁ F₁] :
    (functorHomEquiv V) (𝟙 F₁) = functorEnrichedId V F₁ := by
  aesop_cat
```

Reply:
Thanks! (I have kept only the `apply_app` lemma, as the other one is not very useful.)